src/core/job.c

```diff
@@ -1393,6 +1393,9 @@ void job_shutdown_magic(Job *j) {
         if (!unit_has_name(j->unit, SPECIAL_SHUTDOWN_TARGET))
                 return;
 
+        /* This is the very beginning of the shutdown phase, so take the timestamp here */
+        dual_timestamp_now(j->unit->manager->timestamps + MANAGER_TIMESTAMP_SHUTDOWN_START);
```

Comment:
this should probably be set for the user manager too. i.e. please switch the two conditions above around and put this in between

Reply:
also, while the usual way to shutdown is to go through the job engine, we also support shutting down immediately (i.e. poweroff -f), which bypasses the job engine. I think we should initialize this timestamp in that case too, but only if it wasn#t set here by the job engine already.

or in other words, in invoke_main_loop() before you do the regular switch, initialize the timestamp if it isn't set up yet, in case the objective is one of soft reboot, poweroff, … and so on.